frontend/src/pages/clusterSettings/PVCSizeSettings.tsx

```diff
@@ -0,0 +1,93 @@
+import * as React from 'react';
+import {
+  Button,
+  ButtonVariant,
+  HelperText,
+  HelperTextItem,
+  InputGroup,
+  InputGroupText,
+  InputGroupTextVariant,
+  Stack,
+  StackItem,
+  TextInput,
+} from '@patternfly/react-core';
+import SettingSection from '~/components/SettingSection';
+import { DEFAULT_PVC_SIZE, MAX_PVC_SIZE, MIN_PVC_SIZE, PVC_SIZE_HINT } from './const';
+
+type PVCSizeSettingsProps = {
+  initialValue: number | string;
+  pvcSize: number | string;
+  setPvcSize: (size: number | string) => void;
+};
+
+const PVCSizeSettings: React.FC<PVCSizeSettingsProps> = ({ initialValue, pvcSize, setPvcSize }) => {
+  React.useEffect(() => {
+    setPvcSize(initialValue);
+  }, [initialValue, setPvcSize]);
+
+  return (
+    <SettingSection
+      title="PVC size"
+      description="Changing the PVC size changes the storage size attached to the new notebook servers for
+all users."
+    >
+      <Stack hasGutter>
+        <StackItem>
+          <InputGroup>
+            <TextInput
+              id="pvc-size-input"
+              style={{ maxWidth: '200px' }}
+              name="pvc"
+              data-id="pvc-size-input"
+              type="text"
+              aria-label="PVC Size Input"
+              value={pvcSize}
+              pattern="/^(\s*|\d+)$/"
```

Comment:
Just a small nit, if we have other regex stored as constants maybe we can add this too right?